uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/component/views/CheckoutView.kt

```diff
@@ -112,6 +114,24 @@ internal class CheckoutView @JvmOverloads constructor(context: Context,
             }
         }
 
+        loyaltyView.delegate = object : LoyaltyContract.LoyaltyViewDelegate {
+            override fun onModeChanged(mode: LoyaltyMode) {
+                if (loyaltyView.getCurrentMode() == LoyaltyMode.ERROR_BAD_CURRENCY) {
+                    loadingButtonCallback.enableButton(false)
```

Comment:
What if the user wants to book a ride but is uninterested if the loyalty component is working. IMHO the errors should prevent the user from burning, not booking (with or without earning where possible)